PLG-Motion | Contextual Data Platform Installation Tutorial (#1054)
arangodb/docs-hugo: site/content/contextual-data-platform/tutorials/evaluate-locally.md

  follow-up tutorials
---
In this tutorial, we install ArangoDB and the Arango Contextual Data Platform
on our own machine, going from an empty local Kubernetes cluster to a running
platform that we can open in the browser. Along the way we meet the tools that
operate every Arango Platform deployment: kind for the cluster, kubectl to talk
to it, and Helm to install the Operator and the platform itself.

By the end, you will have:

- A local Kubernetes cluster running on your machine.
- The ArangoDB Kubernetes Operator and an ArangoDB deployment.
- The Contextual Data Platform installed from the chart you downloaded and
  reachable in a browser.
- A running platform you can build on in the follow-up tutorials, plus a simple
  way to remove everything when you are done.

## Step 1: Check the system requirements

Confirm your machine can run the platform locally and that the command-line
tools you need are installed. You also gather the two things provided by the
Arango team: your license credentials and the chart you downloaded from the
website.

Your machine should have:

- An **x86-64 (amd64)** or **64-bit ARM (arm64)** CPU with at least 4 cores.
- At least 16 GB of RAM.

> - At least 50 GB of free disk space for the container images.

Reviewer: Should we make this higher?
aka. do we have some wiggle room to actually add data?

The platform runs around a dozen pods on a single node, so give your container
runtime enough memory. If you use **Docker Desktop**, open its
**Settings > Resources** and raise the memory limit to at least 16 GiB
(20 GiB is comfortable). The default is often too low, which leaves pods stuck
`Pending` later.

Install the following tools:

- [Docker](https://docs.docker.com/get-docker/) or another container runtime
  that kind supports. kind runs the Kubernetes cluster inside containers.
- [kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation) to create
  the local cluster.
- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) to talk to the
  cluster.
- [Helm](https://helm.sh/docs/intro/install/), version 3.10.3 or newer (Helm 4
  works too), to install the Operator and the platform chart.

If a tool is missing, install it before continuing; the rest of this tutorial
assumes all prerequisites are in place.

Finally, make sure you have the two items from the Arango team ready on disk:

- Your **license credentials**, composed of a client ID and a client secret.
- The **platform chart** you downloaded from the website, a file named
  `chart.tgz`.

> {{< info >}}

Reviewer: I validated on x86-64 as well.

Can we make this a "tab" selection?
So a client clicks somewhere to say which version they are on, so they get the correct commands.

Reviewer: Alternative: Can we get away with making this an environment variable and have the user execute same one liner hierarchy picking up the right one so everything down would reference the variable?

This tutorial was validated on an Apple Silicon (arm64) Mac, so the examples use
`arm64`. On an x86-64 machine, replace `arm64` with `amd64` in Step 4 and
Step 5.
{{< /info >}}

> ## Step 2: Set up a local cluster

Reviewer: Should we make this step optional?
Client has the Alternative to use a running Kubernetes cluster.

Or would this be a different tutorial?

Create a throwaway Kubernetes cluster on your own machine:

```sh
kind create cluster --name arango-platform
```

kind automatically points kubectl at the new cluster, so you don't have to
configure anything by hand. Confirm that a single-node cluster is up and that
kubectl is talking to it:

```sh
kubectl cluster-info --context kind-arango-platform
kubectl get nodes
```

The node should eventually report a status of `Ready`.

> 

Reviewer: I think. we need to add a section for a storage here.
e.g. minio.

Otherwise the FileUpload will not work, and therefore the AI Suite cannot work.

  --namespace arango \
  "https://github.com/arangodb/kube-arangodb/releases/download/${VERSION_OPERATOR}/kube-arangodb-enterprise-${VERSION_OPERATOR}.tgz" \
  --set "webhooks.enabled=true" \
  --set "operator.args[0]=--deployment.feature.gateway=true" \
  --set "operator.architectures={arm64}"
```

The output looks similar to the following on success:

```
Release "operator" does not exist. Installing it now.
NAME: operator
LAST DEPLOYED: Thu Jul  9 12:52:39 2026
NAMESPACE: arango
STATUS: deployed
REVISION: 1
DESCRIPTION: Install complete
TEST SUITE: None
NOTES:
You have installed Kubernetes ArangoDB Operator in version 1.4.2
```

Wait for the Operator pod to become ready and verify it is running:

```sh
kubectl wait --for=condition=ready pod --selector app.kubernetes.io/name=kube-arangodb-enterprise --namespace arango --timeout=120s

kubectl get pods --namespace arango --selector app.kubernetes.io/name=kube-arangodb-enterprise
```

Expected output (`x` stands for a varying letter or digit):

```
NAME                                        READY   STATUS    RESTARTS   AGE
arango-operator-operator-xxxxxxxxxx-xxxxx   1/1     Running   0          45s
```

> 

Reviewer: Please add a section here:
If the above does not succeed, and you are hitting a timeout, please check again if you have picked the desired architecture.
The above command will get stuck if you try to install an arm64 operator, but only have amd64 machines in your cluster.
You can see this with this command:

` kubectl get deployments arango-operator-operator --namespace arango -o yaml`

Note: There is a status section which has an Error message in it.

## Step 5: Create a deployment

Declare a small, single-server ArangoDB deployment sized for a local machine.
You give the deployment a name you reuse when you install the platform; this
tutorial names it `deployment`.

The gateway feature is enabled so the platform can expose all of its services
through a single port, and vector indexes are enabled because they are required
by features such as GraphRAG. The `license` field points at the secret you
created in Step 3, and `architecture` must match your CPU (`arm64` or `amd64`).

Save the following specification as `deployment.yaml`:

> ```yaml

Reviewer: This is actually worse, it contains arm64 again.
if you are evaluating on amd64 this will get stuck.

Note:
In the "stuck" case the `--watch` did never make progress.

Also add the way out:

In case you are getting stuck there is a chance you missed changing the architecture here as well.
Here is how to figure this out:

```
kubectl get pods --namespace arango
NAME                                        READY   STATUS    RESTARTS   AGE
arango-operator-operator-5df76999cf-hsqb5   1/1     Running   0          17m
deployment-id-19a661                        0/1     Pending   0          14m
```

This will list your pods, one item the operator you had installed previously, the user is the Arango deployment, in my case `deployment-id-19a661` the last digits are generated and will be different for you.

As you see it is "Pending" for quite a while.
You can access details with:

`kubectl get pods --namespace arango deployment-id-19a661 -o yaml`

The relevant part is at the end of the output:

```
[...]
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2026-07-13T14:07:48Z"
    message: '0/1 nodes are available: 1 node(s) didn''t match Pod''s node affinity/selector.
      preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling.'
    reason: Unschedulable
    status: "False"
    type: PodScheduled
  phase: Pending
  qosClass: BestEffort
```

`message: '0/1 nodes are available: 1 node(s) didn''t match Pod''s node affinity/selector.`
Indicates we cannot start anything on the given architecture.

Change the downloaded file above and look for `architecture:` there make sure you insert arm64 or amd64 whichever is the desired one.

Then repeat:
`kubectl apply --namespace arango -f deployment.yaml`

This now changes the desired state to the correct one, now we need to kill the stuck pod:

`kubectl delete pods --namespace arango deployment-id-19a661`
(note the ID needs to be matching with yours again)

Now the pod is gone, but we defined that such a pod needs to exist, so the operator will try to spawn it again, now picking up the changed architecture.

apiVersion: "database.arangodb.com/v1"
kind: "ArangoDeployment"
metadata:
  name: "deployment"
spec:

>   mode: Single

Reviewer: Do we want to show single server?
Or should we go to Cluster, which is the default for a user.

AFAIK it should not matter for the tools

  image: "arangodb/enterprise:3.12.9"
  architecture:
    - arm64
  gateway:
    enabled: true
    dynamic: true
  gateways:
    count: 1
  single:
    args:

>       - --vector-index

Reviewer: Very good this is important :+1:

  license:
    secretName: arango-license-key
```

Apply the specification and watch the pods start:

```sh
kubectl apply --namespace arango -f deployment.yaml

kubectl get pods --namespace arango --watch  # Ctrl+C to stop watching
```

After a few minutes, a database pod (`deployment-sngl-*`) and a gateway pod
(`deployment-gway-*`) reach the `Running` state. The first start is slow because
container images are being pulled; this is expected and only happens once.

## Step 6: Install the Contextual Data Platform chart

Install the platform from the `chart.tgz` file you downloaded from the website.
The chart creates the platform's services as custom resources, and the Operator
then pulls each service and runs it.

Run the command from the directory that contains `chart.tgz`. The
`--set deployment=deployment` option tells the chart which ArangoDB deployment
to attach to; the value must match the `metadata.name` of the `ArangoDeployment`
from Step 5:

```sh

> helm upgrade platform --install ./chart.tgz \

Reviewer: Side note: Maybe we should make a "non-gpu" variant a release artifact.
The gpu images are super large, and only relevant if I want to host my own training method or LLM hoster.
If I connect to OpenAI anyhow it does not matter.

  --namespace arango \
  --set deployment=deployment
```

On success, Helm reports the release as deployed and lists the platform charts
and services it created:

```
Release "platform" does not exist. Installing it now.
NAME: platform
NAMESPACE: arango
STATUS: deployed
REVISION: 1
NOTES:
Arango Platform Release has been installed!

Charts:
ArangoPlatformChart arango-control-plane in Version v0.0.21
...

Services:
ArangoPlatformService arangodb-platform-ui
...
```

You do not edit the chart; you simply point Helm at it.

## Step 7: Verify the installation

The Operator now downloads and starts each platform service. Watch the pods
until they are all up:

```sh

> kubectl get pods --namespace arango --watch  # Ctrl+C to stop watching

Reviewer: Ah okay, --watch will only update on change.
So waiting 5 minutes without a change (e.g. in stuck state) will look like nothing is happening.

Reviewer: Still I am not too happy with --watch UX.

it is nice to see something happening but it never gives the "I am done" feedback.

file-manager-xxxxxxxxx-xxxxx                            2/2     Running   0          16m
platform-monitoring-grafana-xxxxxxxxx-xxxxx             2/2     Running   0          16m
platform-monitoring-prometheus-server-xxxxxxxxx-xxxxx   3/3     Running   0          16m
```

You can also check the higher-level platform services, which should all report
`READY` as `True`:

```sh
kubectl get arangoplatformservices --namespace arango
```

If a pod stays in `Pending` or keeps restarting, the machine most likely ran out
of resources; revisit the memory advice in Step 1. To see why a specific pod is
not starting, describe it and check its events and logs:

```sh
kubectl describe pod <pod-name> --namespace arango
kubectl logs <pod-name> --namespace arango
```

## Step 8: Open the web interface

The platform exposes all of its services through the gateway on port `8529`
inside Kubernetes. Forward that port to your machine:

```sh
kubectl port-forward --namespace arango \
  service/deployment-ea 8529:8529
```

Leave that command running and open the unified web interface in your browser:

<https://127.0.0.1:8529/ui/>

> 

Reviewer: You should mention that the default username and password are `root` with empty password.

And name that this is not secure for production.

Your browser warns about a self-signed certificate; for a local setup this is
expected, so continue past it. Depending on your browser, you may need to click
a button for advanced options first.

The Contextual Data Platform web interface loads, and the platform you installed
is now something you can see and use. This is the payoff moment of the whole
tutorial.

Stop the port forwarding when you are done by pressing {{< kbd "Ctrl C" >}} in
the terminal that runs it.

## What's next

You now have a healthy Arango Contextual Data Platform running on your machine
and reachable in the browser. This is the launch pad for the tutorials that
follow, which pick up exactly where this one ends:

- Explore the core database: load data and query it.

> - Build contextual retrieval with AutoGraph and GraphRAG.

Reviewer: afaik official wording is just "AutoRAG" but I am no authority here.

---
title: Get started with the Arango Contextual Data Platform locally
menuTitle: Evaluate locally
weight: 5
description: >-
  A hands-on tutorial that takes you from an empty machine to a running
  Arango Contextual Data Platform you can open in your browser, ready for the
  follow-up tutorials
---
In this tutorial, we install ArangoDB and the Arango Contextual Data Platform
on our own machine, going from an empty local Kubernetes cluster to a running
platform that we can open in the browser. Along the way we meet the tools that
operate every Arango Platform deployment: kind for the cluster, kubectl to talk
to it, and Helm to install the Operator and the platform itself.

By the end, you will have:

- A local Kubernetes cluster running on your machine.

> - The ArangoDB Kubernetes Operator and an ArangoDB deployment.

Reviewer: ```suggestion
- The Arango Kubernetes Operator and an Arango.ai deployment.
```

Should we go to new naming here?
You are getting more than the Database.

pods stuck `Pending` later.

Install the following tools:

- [Docker](https://docs.docker.com/get-docker/) or another container runtime
  that kind supports. kind runs the Kubernetes cluster inside containers.
- [kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation) to create
  the local cluster.
- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) to talk to the
  cluster.
- [Helm](https://helm.sh/docs/intro/install/), version 3.10.3 or newer (Helm 4
  works too), to install the Operator and the platform chart.

If a tool is missing, install it before continuing; the rest of this tutorial
assumes all prerequisites are in place.

Finally, make sure you have the two items from the Arango team ready on disk:

- Your **license credentials**, composed of a client ID and a client secret.
- The **platform chart** you downloaded from the website, a file named
  `chart.tgz`.

{{< info >}}
This tutorial was validated on both Apple Silicon (arm64) and x86-64 (amd64)
machines. In Step 4 you set an `ARCH` variable to match your CPU, and the later
commands read it, so you choose your architecture only once.
{{< /info >}}

## Step 2: Set up a local cluster

> Create a throwaway Kubernetes cluster on your own machine. The `--image` flag

Reviewer: ```suggestion
If you already run a Kubernetes cluster and would rather use it, you can skip
this step, but that path is outside the scope of this tutorial.

Create a throwaway Kubernetes cluster on your own machine. The `--image` flag
```

pins the Kubernetes version this tutorial was validated with, so your cluster
matches the examples that follow:

```sh
kind create cluster --name arango-platform --image kindest/node:v1.33.1
```

If you already run a Kubernetes cluster and would rather use it, you can skip

> this step, but that path is outside the scope of this tutorial.

Reviewer: ```suggestion
```

Moved this BEFORE you actually start the cluster.
If you do not want it, you should jump over the command.

This tutorial uses the name `arango`:

```sh
kubectl create namespace arango
```

## Step 3: Create a secret for the license

Store your license credentials in the cluster as a secret so that the Operator
can activate the deployment and keep the license renewed automatically.

License activation and renewal need ongoing internet access. For the renewal
lifecycle, the required network access (`*.license.arango.ai`), and the
configuration options, see [License Management](../license-management.md).

Substitute `<license-client-id>` and `<license-client-secret>` with your actual
license credentials. Keep the whole command on one logical line (the backslashes
continue it):

```sh
kubectl create secret generic arango-license-key \
  --namespace arango \
  --from-literal=license-client-id="<license-client-id>" \
  --from-literal=license-client-secret="<license-client-secret>"
```

Verify that the secret was created:

```sh
kubectl get secret arango-license-key --namespace arango
```

Expected output:

```
NAME                 TYPE     DATA   AGE
arango-license-key   Opaque   2      10s
```

## Step 4: Install the Operator

> Install the [ArangoDB Kubernetes Operator](https://arangodb.github.io/kube-arangodb/)

Reviewer: 🤔 we need to somehow make clear that this link is more a FYI.
You do not actually need to follow it, the install command is following.

Reviewer: I read this line as:

Please now install the operator, here is link with instructions.
But when I click at it, I am a bit lost in the big overview of all technical details I do not really need at this point

(`kube-arangodb`) with Helm. It watches your declarations and creates the
matching Kubernetes resources; nothing database-related runs until you declare
it in the next step. For how the pieces fit together, see the
[architecture overview](../architecture.md).

This tutorial installs and was validated with ArangoDB Kubernetes Operator version `1.4.2`.
Newer releases are listed on [GitHub](https://github.com/arangodb/kube-arangodb/releases/),
but staying on the tested version keeps your output matching the examples below.

First, set an `ARCH` variable to match your CPU. Every command that needs your
architecture reads this variable, so you set it only once here. Then install the
Operator with the options shown to enable webhooks and the gateway feature:

```sh

> ARCH=arm64 # Apple Silicon; use amd64 on x86-64

Reviewer: This one liner will autodetect it
```suggestion
ARCH=$([ "$(uname -m)" = x86_64 ] && echo amd64 || echo arm64)
```

```

The output looks similar to the following on success:

```
Release "operator" does not exist. Installing it now.
NAME: operator
LAST DEPLOYED: Thu Jul  9 12:52:39 2026
NAMESPACE: arango
STATUS: deployed
REVISION: 1
DESCRIPTION: Install complete
TEST SUITE: None
NOTES:
You have installed Kubernetes ArangoDB Operator in version 1.4.2
```

Wait for the Operator pod to become ready and verify it is running:

```sh
kubectl wait --for=condition=ready pod --selector app.kubernetes.io/name=kube-arangodb-enterprise --namespace arango --timeout=120s

kubectl get pods --namespace arango --selector app.kubernetes.io/name=kube-arangodb-enterprise
```

Expected output (`x` stands for a varying letter or digit):

```
NAME                                        READY   STATUS    RESTARTS   AGE
arango-operator-operator-xxxxxxxxxx-xxxxx   1/1     Running   0          45s
```

If `kubectl wait` times out and the Operator pod never becomes ready, it was
most likely installed for a different architecture than your nodes provide (for
example, an `arm64` Operator on `amd64` machines). Check the deployment status:

```sh
kubectl get deployment arango-operator-operator --namespace arango -o yaml
```

> The `status` section reports an error about unschedulable pods or a node

Reviewer: Here is an example of the broken output:

status:
  conditions:
  - lastTransitionTime: "2026-07-21T09:14:13Z"
    lastUpdateTime: "2026-07-21T09:14:13Z"
    message: Deployment does not have minimum availability.
    reason: MinimumReplicasUnavailable
    status: "False"
    type: Available
  - lastTransitionTime: "2026-07-21T09:14:13Z"
    lastUpdateTime: "2026-07-21T09:14:13Z"
    message: ReplicaSet "arango-operator-operator-6cf6c64c9f" is progressing.
    reason: ReplicaSetUpdated
    status: "True"

This is the important part:
 message: Deployment does not have minimum availability.
    reason: MinimumReplicasUnavailable

Declare an ArangoDB cluster deployment. You give the deployment a name you reuse
when you install the platform; this tutorial names it `deployment`.

The gateway feature is enabled so the platform can expose all of its services
through a single port, and vector indexes are enabled on the DB-Servers and
Coordinators because they are required by features such as GraphRAG. The
`license` field points at the secret you created in Step 3, and the deployment
reads the `ARCH` variable from Step 4, so you do not set the architecture by
hand here.

Apply the specification directly so the `ARCH` variable is filled in. This
deployment uses three DB-Servers and two Coordinators:

```sh
cat <<EOF | kubectl apply --namespace arango -f -
apiVersion: "database.arangodb.com/v1"
kind: "ArangoDeployment"
metadata:
  name: "deployment"
spec:
  mode: Cluster
  image: "arangodb/enterprise:3.12.9"
  architecture:
    - ${ARCH}
  gateway:
    enabled: true
    dynamic: true
  gateways:
    count: 1
  dbservers:
    count: 3
    args:
      - --vector-index
  coordinators:
    count: 2
    args:
      - --vector-index
  license:
    secretName: arango-license-key
EOF
```

Watch the pods start:

```sh

> kubectl get pods --namespace arango --watch  # Ctrl+C to stop watching

Reviewer: This comment applies to everything with `--watch`

This is just a suggestion and no hard requirement.

Should we instead use:

`watch -n 1 kubectl get pods --namespace arango`

This will display the full output, and update in place.
Then you can actually see: All are "running" and stop.

Like this (from later state, only figured this out there.)

arangodb-core-ui-85457446b-sbxs9                        2/2     Running   0          88s
Every 1.0s: kubectl get pods --namespace arango                                                                                       mchacki-office: Tue Jul 21 11:34:09 2026

NAME                                                    READY   STATUS    RESTARTS   AGE
arango-control-plane-6f4ccc5d7f-dqglq                   3/3     Running   0          5m42s
arango-operator-operator-5df76999cf-6zsr7               1/1     Running   0          16m
arangodb-core-ui-85457446b-sbxs9                        2/2     Running   0          5m31s
arangodb-mlflow-0                                       2/2     Running   0          5m46s
arangodb-platform-ui-5745d7498b-kfjls                   2/2     Running   0          5m50s
arangodb-platform-ui-server-7b75bfc875-spzfw            2/2     Running   0          5m51s
deployment-agnt-74vkq75e-a5d801                         1/1     Running   0          14m
deployment-agnt-na4q6wyy-a5d801                         1/1     Running   0          14m
deployment-agnt-o2gzjy1y-a5d801                         1/1     Running   0          14m
deployment-crdn-u0yjwrs2-a5d801                         1/1     Running   0          14m
deployment-crdn-xrhcgry9-a5d801                         1/1     Running   0          14m
deployment-gway-igaupx0m-a5d801                         2/2     Running   0          13m
deployment-ml-0                                         2/2     Running   0          5m45s
deployment-prmr-btv7h4fp-a5d801                         1/1     Running   0          14m
deployment-prmr-hyjl4t9n-a5d801                         1/1     Running   0          14m
deployment-prmr-lewmehz7-a5d801                         1/1     Running   0          14m
file-manager-7d9fb9b6d9-sctr4                           2/2     Running   0          5m50s
platform-monitoring-grafana-7678f4855d-v8n5q            2/2     Running   0          5m34s
platform-monitoring-prometheus-server-97945c8dd-qw8x9   3/3     Running   0          5m40s

```sh
kubectl describe pods --namespace arango --selector app.kubernetes.io/name=kube-arangodb-enterprise
```

The `Events` section shows a node affinity or `nodeSelector` mismatch. If so,
correct the `ARCH` value and rerun the `helm upgrade` command above.

## Step 5: Create a deployment

Declare an ArangoDB cluster deployment. You give the deployment a name you reuse
when you install the platform; this tutorial names it `deployment`.

The gateway feature is enabled so the platform can expose all of its services
through a single port, and vector indexes are enabled on the DB-Servers and
Coordinators because they are required by features such as GraphRAG. The
`license` field points at the secret you created in Step 3, and the deployment
reads the `ARCH` variable from Step 4, so you do not set the architecture by
hand here.

Apply the specification directly so the `ARCH` variable is filled in. This
deployment uses three DB-Servers and two Coordinators:

```sh
cat <<EOF | kubectl apply --namespace arango -f -
apiVersion: "database.arangodb.com/v1"
kind: "ArangoDeployment"
metadata:
  name: "deployment"
spec:
  mode: Cluster

>   image: "arangodb/enterprise:3.12.9"

Reviewer: Can we update this to 3.12.10

as we had a new release

🤔 should we just update this to "latest" ?
This way we have no fixed version number in the docs and do not need to remember to update this.

>   image: "arangodb/enterprise:3.12.9"

Author: ```suggestion
  image: "arangodb/enterprise:latest"
```
